cypress/support/metaTestHelper.js

```diff
@@ -81,31 +81,38 @@ export const retrieve404BodyResponse = (url, bodyResponse) => {
     .should('include', bodyResponse);
 };
 
+export const retrieveData = data => {
+  const description = data.promo.summary;
+  const title = data.promo.headlines.seoHeadline;
+  const { type } = data.metadata;
+  const firstPublished = new Date(data.metadata.firstPublished).toISOString();
+  const lastPublished = new Date(data.metadata.lastPublished).toISOString();
+
+  retrieveMetaDataContent('head meta[name="description"]', description);
+  retrieveMetaDataContent('head meta[name="og:title"]', title);
+  retrieveMetaDataContent('head meta[name="og:type"]', type);
+  retrieveMetaDataContent(
+    'head meta[name="article:published_time"]',
+    firstPublished,
+  );
+  retrieveMetaDataContent(
+    'head meta[name="article:modified_time"]',
+    lastPublished,
+  );
+};
 export const metadataAssertion = () => {
   cy.window().then(win => {
     const windowData = win.SIMORGH_DATA.data;
-    const description = windowData.promo.summary;
     const { language } = windowData.metadata.passport;
-    const title = windowData.promo.headlines.seoHeadline;
-    const { type } = windowData.metadata;
-    const firstPublished = new Date(
-      windowData.metadata.firstPublished,
-    ).toISOString();
-    const lastPublished = new Date(
-      windowData.metadata.lastPublished,
-    ).toISOString();
-
-    retrieveMetaDataContent('head meta[name="description"]', description);
-    retrieveMetaDataContent('head meta[name="og:title"]', title);
-    retrieveMetaDataContent('head meta[name="og:type"]', type);
-    retrieveMetaDataContent(
-      'head meta[name="article:published_time"]',
-      firstPublished,
-    );
-    retrieveMetaDataContent(
-      'head meta[name="article:modified_time"]',
-      lastPublished,
-    );
+    retrieveData(windowData);
     getElement('html').should('contain', language);
   });
 };
+
+export const metadataAssertionAMP = service => {
```

Comment:
We'll want to check the `lang` attribute exists on AMP pages as well as canonical pages, so I believe you can use just `metadataAssertion` and not need `metadataAssertionAMP`.